README.md

```diff
@@ -37,7 +35,56 @@ Please see https://github.com/flojoy-io/flojoy-desktop/tree/main/PYTHON/FUNCTION
 See https://github.com/flojoy-io/flojoy-python/issues/4
 
 
+## Reconcilers
+
+Flojoy nodes should try to accomodate any reasonable combination of inputs that a first-time Flojoy Studio user might try.
+
+For example, the ADD node should make a best effort to do something reasonable when a matrix is added to a dataframe, or a 2 matrices of a different size are added.
+
+For this reason, we've created the `Reconciler` class to handle the process of turning different data types into compatible, easily added objects. 
+
+### An Example
+
+Here's an example of the Reconciler in action from the ADD node. Just initialize the Reconciler and pass it pairs of primitives from within your DataContainers.
+
+```python
+def ADD(dc_inputs: list[DataContainer], params: dict) -> DataContainer:
+  ...
+    
+    dc_reconciler = Reconciler()
+    cur_sum = dc_inputs[0]
+    for operand_n in dc_inputs[1:]:
+        # reconcile the types of the inputs before calling numpy
+        reconciled_lhs, reconciled_rhs = dc_reconciler.reconcile(cur_sum.y, operand_n.y)
```

Comment:
This is great! Can you imagine user needing to customize reconcile behavior? We could take in optional arguments to support those

Reply:
@dev-nested for sure! I suspect there'll be regimes of nodes that will reconcile in non-obvious ways.(Signal / Image processing for example)